src/components/NewGalaxyModal.js

```diff
@@ -96,32 +143,70 @@ export const NewGalaxyModal = _.flow(
   }
 
   const renderDeleteWarning = () => {
-    return div({ style: { lineHeight: '22px' } }, [
-      div({ style: { marginTop: '0.5rem' } }, [
-        'Deleting your Cloud Environment will also ',
-        span({ style: { fontWeight: 600 } }, ['delete any files on the associated hard disk']),
-        ' (e.g. results files). Double check that your workflow results were written to ',
-        'the data tab in the workspace before clicking “Delete”'
-      ]),
-      div({ style: { marginTop: '1rem' } }, [
-        'Deleting your Cloud Environment will stop your ',
-        'running Galaxy application and your application costs. You can create a new Cloud Environment ',
-        'for Galaxy later, which will take 8-10 minutes.'
+    return h(Fragment, [
+      h(TitleBar, {
+        title: 'Delete Cloud Environment for Galaxy',
+        style: { marginBottom: '0.5rem' },
+        onDismiss,
+        onPrevious: !!viewMode ? () => setViewMode(undefined) : undefined
+      }),
+      div({ style: { lineHeight: '22px' } }, [
+        div({ style: { marginTop: '0.5rem' } }, [
+          'Deleting your Cloud Environment will also ',
+          span({ style: { fontWeight: 600 } }, ['delete any files on the associated hard disk']),
+          ' (e.g. results files). Double check that your workflow results were written to ',
+          'the data tab in the workspace before clicking “Delete”'
+        ]),
+        div({ style: { marginTop: '1rem' } }, [
+          'Deleting your Cloud Environment will stop your ',
+          'running Galaxy application and your application costs. You can create a new Cloud Environment ',
+          'for Galaxy later, which will take 8-10 minutes.'
+        ])
       ])
     ])
   }
 
   const renderLaunchWarning = () => {
-    return div({ style: { lineHeight: '22px' } }, [
-      h(GalaxyWarning)
+    return h(Fragment, [
+      h(TitleBar, {
+        title: h(WarningTitle, ['Launch Galaxy']),
+        style: { marginBottom: '0.5rem' },
+        onDismiss,
+        onPrevious: !!viewMode ? () => setViewMode(undefined) : undefined
+      }),
+      div({ style: { lineHeight: '22px' } }, [
+        h(GalaxyWarning)
+      ])
     ])
   }
 
   const renderDefaultCase = () => {
     const { cpu, memory } = _.find({ name: 'n1-standard-8' }, machineTypes)
     const cost = getGalaxyCost(app || { kubernetesRuntimeConfig: { machineType: 'n1-standard-8', numNodes: 2 } })
     return h(Fragment, [
-      div([`Environment ${app ? 'consists' : 'will consist'} of an application and cloud compute.`]),
+      h(TitleBar, {
+        title: Utils.cond(
```

Comment:
This cond and the cond below are almost identical in a lot of ways, I wonder if a helper function could be extracted to reduce the repeated code